openlibrary/plugins/upstream/mybooks.py

```diff
@@ -272,6 +294,64 @@ def render(
             elif self.key == 'list':
                 docs = list
 
+            elif is_logged_in_user and self.key == "loan_history":
+                account = OpenLibraryAccount.get(username=logged_in_username)
+                s3_keys = web.ctx.site.store.get(account._key).get('s3_keys')
+                limit = RESULTS_PER_PAGE
+                offset = page * limit - limit
+
+                loan_history = s3_loan_api(
+                    # s3_keys=s3_keys, action='user_borrow_history'
+                    s3_keys=s3_keys,
+                    action='user_borrow_history',
+                    limit=limit + 1,
+                    offset=offset,
+                    # newest=True,
+                    # sort=True,
+                ).json()['history']['items']
+
+                show_next = len(loan_history) == limit + 1
+
+                # Drop 'extra' item (used to determine show_next) from results.
+                loan_history = sorted(loan_history, key=lambda k: k.get('identifier'))
```

Comment:
```suggestion
```
With the results coming back in reverse chronological order from the API, this should no longer be necessary.